haskell/test/SMT/Common/Run.hs

```diff
@@ -0,0 +1,219 @@
+  -> (Maybe ModelError, Storage, [FA2.TransferParams], Mutez, [Text])
+  -> EmulatedT PureM ()
+printResult mc
+  (haskellErrors, haskellStore, haskellGovStore, haskellGovBalance, haskellViewStore)
+  (ligoErrors, ligoStore, ligoGovStore, ligoGovBalance, ligoViewStore) = do
+    when (haskellErrors /= ligoErrors) $
+      error
+        (  "*** Error: Haskell and Ligo errors are different:\n"
+        <> modelCallMsg
+        <> "* Haskell error:\n"
+        <> show haskellErrors <> "\n"
+        <> "* Ligo error:\n"
+        <> show ligoErrors <> "\n"
+        )
+
+    when (haskellStore /= ligoStore) $
+      error
+        (  "*** Error: Haskell and Ligo storage are different:\n"
+        <> modelCallMsg
+        <> "* Haskell storage:\n"
+        <> pretty haskellStore <> "\n"
+        <> "* Ligo storage:\n"
+        <> pretty ligoStore <> "\n"
+        )
+
+    when (haskellGovStore /= ligoGovStore) $
+      error
+        (  "*** Error: Haskell and Ligo governance contract storage are different:\n"
+        <> modelCallMsg
+        <> "* Haskell governance contract storage:\n"
+        <> pretty haskellGovStore <> "\n"
+        <> "* Ligo governance contract storage:\n"
+        <> pretty ligoGovStore <> "\n"
+        )
+
+    -- Governance contract balance could be updated via treasury/registry transfer proposal.
+    when (haskellGovBalance /= ligoGovBalance) $
+      error
+        (  "*** Error: Haskell and Ligo governance contract balance are different:\n"
+        <> modelCallMsg
+        <> "* Haskell governance contract balance:\n"
+        <> pretty haskellGovBalance <> "\n"
+        <> "* Ligo governance contract balance:\n"
+        <> pretty ligoGovBalance <> "\n"
+        )
+
+    when (haskellViewStore /= ligoViewStore) $
+      error
+        (  "*** Error: Haskell and Ligo view contract storage are different:\n"
+        <> modelCallMsg
+        <> "* Haskell view contract storage:\n"
+        <> pretty haskellViewStore <> "\n"
+        <> "* Ligo view contract storage:\n"
+        <> pretty ligoViewStore <> "\n"
+        )
+
+    where
+      modelCallMsg = "* Call with:\n" <> (pretty mc)
+
+runBaseDaoLigo
+  :: (TAddress Parameter, TAddress FA2.Parameter, TAddress (MText, Maybe MText))
+  -> ModelCall
+  -> EmulatedT PureM (Maybe ModelError, FullStorage, [FA2.TransferParams], Mutez, [Text])
+runBaseDaoLigo (dao, gov, viewC) mc = do
+  case (mc & mcAdvanceLevel) of
+    Just lvl -> advanceLevel lvl
+    Nothing -> pure ()
+
+  nettestResult <- attempt @NettestFailure $ callLigoEntrypoint mc dao
+  let result = parseNettestError nettestResult
+  fs <- getFullStorage (unTAddress dao)
+
+  govStore <- fromVal @[FA2.TransferParams] <$> getStorage' @(ToT [FA2.TransferParams]) gov
+  govBalance <- getBalance gov
+
+  viewStorage <- fromVal @[(MText, Maybe MText)] <$> getStorage' @(ToT [(MText, Maybe MText)]) viewC
+  pure (result, fs, govStore, govBalance, show <$> viewStorage)
```

Comment:
As another point to the documentation, it's almost impossible to tell what are these values for without looking at their usage.

This makes it hard to spot mistakes or possible improvements.

For example, I'm pretty sure that instead of returning a `Maybe ModelError` and a `FullStorage` we could return an `Either ModelError FullStorage` for the same comment about the fact that the storage doesn't change when there is an error (so it doesn't need to be checked) and vice-versa. However it's pretty hard to tell if that's correct.

Reply:
I've used `Maybe ModelError` since it is the type of the result from Nettest `attempt`, and `runBasedaoHaskell` (now renamed to `handleCallViaHaskell`).

It does make more sense to use `Either`, it just requires some wrap and unwrapping. Nevertheless, I've updated this to use `Either`.

Also added comment to explain what the values are.